redash/query_runner/uptycs.py

```diff
@@ -0,0 +1,141 @@
+from redash.utils import JSONEncoder
+
+enabled = True
+
+urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+class Uptycs(BaseSQLQueryRunner):
+
+    noop_query = "SELECT 1"
+
+    @classmethod
+    def configuration_schema(cls):
+        return {
+            "type": "object",
+            "properties": {
+                "url": {
+                    "type": "string"
+                },
+                 "customerid": {
+                    "type": "string"
+                },
+                "key": {
+                    "type": "string"
+                },
+                "secret": {
+                    "type": "string",
+                },
+            },
+            "order": ['url', 'customerid', 'key', 'secret'],
+            "required": ["url", "customerid", "key","secret"],
+            "secret": ["secret", "customerid", "key"]
+        }
+
+
+    @classmethod
+    def enabled(cls):
+        return enabled
+
+    @classmethod
+    def annotate_query(cls):
+        return False
+
+
+    @classmethod
+    def type(cls):
+        return "uptycs"
+
+
+    def generateHeaders(self, key, secret):
+        header={}
+        utcnow=datetime.datetime.utcnow()
+        date=utcnow.strftime("%a, %d %b %Y %H:%M:%S GMT")
+        authVar = jwt.encode({'iss': key}, secret, algorithm='HS256')
+        authorization="Bearer %s" % (authVar)
+        header['date']=date
+        header['Authorization']=authorization
+        return header
+
+    def transformed_to_redash_json(self, data):
+        transformed_columns = []
+        rows = []
+        # convert all type to JSON string
+        # In future we correct data type  mapping later
+        if 'columns' in data:
+            for json_each in data['columns']:
+                name = json_each['name']
+                new_json = {"name": name, 
+                            "type": "string", 
+                            "friendly_name": "null"}
+                transformed_columns.append(new_json)
+        # Transfored items into rows.
+        if 'items' in data:
+            rows = data['items']
+
+        redash_json_data = {"columns": transformed_columns, 
+                            "rows": rows}  
+        return redash_json_data
+
+    def restcall(self, sql):
+        # JWT encoded header
+        header = self.generateHeaders( self.configuration.get('key'), 
+                                       self.configuration.get('secret'))
+
+        # URL form using API key file based on GLOBAL
+        url=("%s/public/api/customers/%s/query" %
+            (self.configuration.get('url'), 
+             self.configuration.get('customerid')));
+
+        # post data base sql
+        post_data_json = { "query": sql }
+
+        response = requests.post(url, headers=header, 
+                json=post_data_json, verify=False )
+        response_output = json.loads(response.content)
+     
+        json_data = self.transformed_to_redash_json(response_output)
```

Comment:
It's better to check for error before transforming the results and only assign value to json_data if there was no error.